src/qubic.cpp

```diff
@@ -3426,11 +3490,67 @@ static bool loadAllNodeStates()
 
 #endif
 
-static void tickProcessor(void*)
-{
-    enableAVX();
-    unsigned long long processorNumber;
-    mpServicesProtocol->WhoAmI(mpServicesProtocol, &processorNumber);
+
+// Calculates computor indices during epoche changes.
+// Requalifying computors keept their index
+// New computor obtain one of the free indices
+ // #pragma optimize("", off)  // Turn off all optimizations
+static void calculateComputorIndex(){
+    
+    // Use reorg buffer 
+    m256i* tempComputorList = (m256i*)reorgBuffer;
+    bool* isIndexTaken = (bool*)(tempComputorList + (NUMBER_OF_COMPUTORS * sizeof(m256i)));  // Start right after tempComputorList
+    bool* isFComputorUsed = (bool*)(isIndexTaken + NUMBER_OF_COMPUTORS);  // Start after isIndexTaken
+
+    bs->SetMem(tempComputorList, NUMBER_OF_COMPUTORS * sizeof(m256i), 0);
+    bs->SetMem(isIndexTaken, NUMBER_OF_COMPUTORS, false);
+    bs->SetMem(isFComputorUsed, NUMBER_OF_COMPUTORS, false);
+
+
+    // Check if computor is keeping it's status and keep it's index
+    for (unsigned int i = 0; i < NUMBER_OF_COMPUTORS; i++) {
+        for (unsigned int j = 0; j < NUMBER_OF_COMPUTORS; j++) {
+            if (system.futureComputors[i] == broadcastedComputors.computors.publicKeys[j]) {
+                // Keep index
+                tempComputorList[j] = system.futureComputors[i];
+                isIndexTaken[j] = true;
+                isFComputorUsed[i] = true;
+                break;
+            }
+        }
+    }
+
+    // Fill remaining spots
+    unsigned int futureComputorIdx = 0;
+    for (unsigned int i = 0; i < NUMBER_OF_COMPUTORS; i++) {
+        if (isIndexTaken[i] == 0) {

```

Comment:
isIndexTaken[i] is bool, better: either !isIndexTaken[i] or isIndexTaken[i] == false